packages/@expo/config-plugins/src/utils/__tests__/plugin-resolver-test.ts

```diff
@@ -1,22 +1,71 @@
-  it('file path', () => {
-    expect(moduleNameIsDirectFileReference('./app')).toBe(true);
-    expect(moduleNameIsDirectFileReference('~/app')).toBe(true);
-    expect(moduleNameIsDirectFileReference('/app')).toBe(true);
-    expect(moduleNameIsDirectFileReference('.')).toBe(true);
-  });
-  it('module', () => {
-    expect(moduleNameIsDirectFileReference('app')).toBe(false);
-    expect(moduleNameIsDirectFileReference('@expo/app')).toBe(false);
-  });
-  it('module folder', () => {
-    expect(moduleNameIsDirectFileReference('app/')).toBe(true);
-    expect(moduleNameIsDirectFileReference('@expo/app/')).toBe(true);
+import { moduleNameIsDirectFileReference, resolvePluginForModule } from '../plugin-resolver';
+jest.unmock('resolve-from');
+
+describe('plugin resolver', () => {
+  describe(moduleNameIsDirectFileReference, () => {
+    it('file path', () => {
+      expect(moduleNameIsDirectFileReference('./app')).toBe(true);
+      expect(moduleNameIsDirectFileReference('~/app')).toBe(true);
+      expect(moduleNameIsDirectFileReference('/app')).toBe(true);
+      expect(moduleNameIsDirectFileReference('.')).toBe(true);
+    });
+    it('module', () => {
+      expect(moduleNameIsDirectFileReference('app')).toBe(false);
+      expect(moduleNameIsDirectFileReference('@expo/app')).toBe(false);
+    });
+    it('module folder', () => {
+      expect(moduleNameIsDirectFileReference('app/')).toBe(true);
+      expect(moduleNameIsDirectFileReference('@expo/app/')).toBe(true);
+    });
+    it('module file', () => {
+      expect(moduleNameIsDirectFileReference('app/index.js')).toBe(true);
+      expect(moduleNameIsDirectFileReference('@expo/app/index')).toBe(true);
+    });
   });
-  it('module file', () => {
-    expect(moduleNameIsDirectFileReference('app/index.js')).toBe(true);
-    expect(moduleNameIsDirectFileReference('@expo/app/index')).toBe(true);
+
+  describe(resolvePluginForModule, () => {
+    const projectRoot = path.resolve(__dirname, 'fixtures');
+
+    it('throws an error for non-existent plugin', () => {
+      expect(() => resolvePluginForModule(projectRoot, './testPlugin__wrong_path.js')).toThrow(
+        `Failed to resolve plugin for module "./testPlugin__wrong_path.js" relative to`
+      );
+    });
+
+    it.each([
+      {
+        pluginReference: './testPlugin.js',
+        expected: { filePath: 'testPlugin.js', isPluginFile: false },
+      },
+      {
+        pluginReference: './testPlugin',
+        expected: { filePath: 'testPlugin.js', isPluginFile: false },
+      },
+      {
+        pluginReference: './node_modules/test-plugin/lib/commonjs/index.js',
+        expected: {
+          filePath: 'node_modules/test-plugin/lib/commonjs/index.js',
+          isPluginFile: false,
+        },
+      },
+      {
+        pluginReference: 'test-plugin',
+        expected: {
+          filePath: 'node_modules/test-plugin/lib/commonjs/index.js',
+          isPluginFile: false,
+        },
+      },
```

Comment:
This test case should throw not found I think. It's mentioned here: https://docs.expo.dev/config-plugins/development-and-debugging/#expo-install

<img width="955" alt="image" src="https://github.com/user-attachments/assets/45bbbfcb-21a4-42f2-bb69-3d65f7b2085e">